json-path/src/main/java/com/jayway/jsonpath/spi/json/JsonOrgJsonProvider.java

```diff
@@ -148,12 +148,16 @@ public boolean isMap(Object obj) {
         JSONObject jsonObject = toJsonObject(obj);
```

Comment:
You could probably simplify this method with streams (Sorry for any syntax errors. I'm writing this in the browser):

```
JSONObject jsonObject  = toJsonObject(obj);
try {
    if(Objects.isNull(jsonObject.names()) return new ArrayList<>();
    
    return jsonObject.names().toList().stream()
        .map(name -> (String) name)
        .collect(Collectors.toList());

} catch(JSONException e) {
// Rest of the code
}
```

Reply:
Yeah, thanks! I have seen all the suggestions you mentioned, and I will change it as soon as possible.

Reply:
I find that the return type of `jsonObject.names()` is `JsonArray`, which doesn't have _method_ like `toList`. But the `jsonObject` has another method `keySet` which return the set of key that is needed. 
(PS: the implementation of `jsonObject.names()` is that it iteratively adds the value in `keySet` into `JsonArray`

Reply:
I find that the return type of `jsonObject.names()` is `JsonArray`, which doesn't have _method_ like `toList`. But the `jsonObject` has another method `keySet` which return the set of key that is needed. 
(PS: the implementation of `jsonObject.names()` is that it iteratively adds the value in `keySet` into `JsonArray`

Reply:
I find that the return type of `jsonObject.names()` is `JsonArray`, which doesn't have _method_ like `toList`. But the `jsonObject` has another method `keySet` which return the set of key that is needed. 
(PS: the implementation of `jsonObject.names()` is that it iteratively adds the value in `keySet` into `JsonArray`

Reply:
https://stleary.github.io/JSON-java/org/json/JSONObject.html#names--
https://stleary.github.io/JSON-java/org/json/JSONArray.html#toList--

JSONObject::names should return a JSONArray
JSONArray should have a method called toList() according to their documentation.

Reply:
Also, I updated the code snippet. Instead of casting (String) name you can do Object::toString

Reply:
Yeah, I find that the version of org.Json in JsonPath is 20140107, where `JSONArray` has no method `toList`.`toList` is introduced in version `20160807`. What could I do?

Reply:
Please check my latest two commits, thanks!
The code of method `.name()` is as follows, I think it is not necessary to use `.name()`. Just using `.keySet()` will be fine.
```
public Iterator keys() {
        return this.keySet().iterator();
    }

public JSONArray names() {
        JSONArray ja = new JSONArray();
        Iterator keys = this.keys();
        while (keys.hasNext()) {
            ja.put(keys.next());
        }
        return ja.length() == 0 ? null : ja;
    }

```

Reply:
Looks good

Reply:
It is my first time to contribute to the open source community, sincerely thanks for your suggestions and approval !